tftrt/examples-cpp/benchmark_runner/main.cpp

```diff
@@ -1,203 +1,228 @@
+    tensorflow::Scope root = tensorflow::Scope::NewRootScope();
+    auto x = tensorflow::ops::Placeholder(root, tensor_host.dtype());
+    auto y = tensorflow::ops::Identity(root, x);
+
+    tensorflow::GraphDef graphDef;
+    TF_RETURN_IF_ERROR(root.ToGraphDef(&graphDef));
+
+    // Create session with identity graph
+    std::unique_ptr<tensorflow::Session> session(
+        tensorflow::NewSession(tensorflow::SessionOptions())
+    );
+    TF_RETURN_IF_ERROR(session->Create(graphDef));
+
+    // Configure to return output on device
+    tensorflow::Session::CallableHandle handle;
+    tensorflow::CallableOptions opts;
+    opts.add_feed("Placeholder:0");
+    opts.add_fetch("Identity:0");
+    opts.mutable_fetch_devices()->insert({"Identity:0", device_name});
+    opts.set_fetch_skip_sync(true);
+    TF_RETURN_IF_ERROR(session->MakeCallable(opts, &handle));
+
+    // Execute graph
+    std::vector<Tensor> tensors_device;
+    Status status = session->RunCallable(handle, {tensor_host}, &tensors_device, nullptr);
+    *tensor_device = tensors_device.front();
+    session->ReleaseCallable(handle);
+    return status;
+}
+
+// Returns info for nodes listed in the signature definition.
+std::vector<tensorflow::TensorInfo> getNodeInfo(
+    const google::protobuf::Map<std::string, tensorflow::TensorInfo>& signature) {
+  std::vector<tensorflow::TensorInfo> info;
+  for (const auto& item : signature) {
+    info.push_back(item.second);
+  }
+  return info;
+}
+
+// Load the `SavedModel` located at `model_dir`.
+Status loadModel(const std::string& model_dir,
+                 tensorflow::SavedModelBundle* bundle,
+                 std::vector<tensorflow::TensorInfo>* input_info,
+                 std::vector<tensorflow::TensorInfo>* output_info) {
+    tensorflow::RunOptions run_options;
+    tensorflow::SessionOptions sess_options;
+    sess_options.config.mutable_gpu_options()->force_gpu_compatible();
+    TF_RETURN_IF_ERROR(tensorflow::LoadSavedModel(sess_options, run_options, model_dir, {"serve"}, bundle));
+
+    // Get input and output names
+    auto signature_map = bundle->GetSignatures();
+    const tensorflow::SignatureDef& signature = signature_map["serving_default"];
+    *input_info = getNodeInfo(signature.inputs());
+    *output_info = getNodeInfo(signature.outputs());
+
+    return Status::OK();
+}
+
+// Create arbitrary inputs matching `input_info` and load them on the device.
+Status setupInputs(const string& device_name,
+                   int32_t batch_size,
+                   std::vector<tensorflow::TensorInfo>& input_info,
+                   std::vector<Tensor>* inputs) {
+    std::vector<Tensor> inputs_device;
+    for (const auto& info : input_info) {
+        auto shape = info.tensor_shape();
+        shape.mutable_dim(0)->set_size(batch_size);
+        Tensor input_host(info.dtype(), shape);
+        Tensor input_device;
+        TF_RETURN_IF_ERROR(moveToDevice(device_name, input_host, &input_device));
```

Comment:
Per Cliff, we want to measure H2D and D2H times as well. Is it possible to instrument those separately?

Reply:
Marked a TODO for this - may want to measure this over repeated runs instead of the single time this done in the current implementation.